File: Backend/LogicPersistence/LogicPersistence.Api/Models/Victim.cs
namespace LogicPersistence.Api.Models;

public class Victim {
	public int id { get; set; }
	public string email { get; set; } = string.Empty;
	public string password { get; set; } = string.Empty;
	public string name { get; set; } = string.Empty;
	public string surname { get; set; } = string.Empty;
	public int prefix { get; set; }
	public int phone_number { get; set; }
	public string address { get; set; } = string.Empty;
	public string identification { get; set; } = string.Empty;

	// FKs
	public int location_id { get; set; }

	//public virtual ICollection<Need> Needs { get; set; } = new List<Need>();
	//public virtual Location Location { get; set; }
	//public virtual ICollection<Donation> Donations { get; set; } = new List<Donation>();
}

File: Desktop/lib/models/victim.dart
class Victim {
  final int id;
  final String email;
  final String name;
  final String surname;
  final int prefix;
  final String phoneNumber;
  final String address;
  final String identification;
  final int locationId;

  Victim({
    required this.id,
    required this.email,
    required this.name,
    required this.surname,
    required this.prefix,
    required this.phoneNumber,
    required this.address,
    required this.identification,
    required this.locationId,
  });

  factory Victim.fromJson(Map<String, dynamic> json) {
    return Victim(
      id: json['id'],
      email: json['email'],
      name: json['name'],
      surname: json['surname'],
      prefix: json['prefix'],
      phoneNumber: json['phone_number'],
      address: json['address'],
      identification: json['identification'],
      locationId: json['location_id'],
    );
  }
}
